### PR TITLE
docs: updated record docs and cli to make golden-test record more clear

### PR DESCRIPTION
**Pull request by capri-xiyue**
updated record docs and cli to make golden-test record more clear

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -919,7 +919,7 @@ steps:
 
 ### Post-rendering validation test (golden test)
 
-We use post-rendering validaton test to record and verify template rendering
+We leverage post-rendering validation tests to record(capture the anticipated outcome akin to expected output in unit test) and subsequently verify template rendering
```
Comment (drevell) [inline README.md:922]: Could you autoformat this file? We're trying to avoid excessive line length.

Comment (drevell) [inline README.md:922]: Needs a space before the parenthesis, lik e`record (capture`

Comment (drevell) [inline README.md:922]: I think the original `use` is better than the updated `leverage`. `Leverage` is unnecessarily complex and abstract.

File: README.md [hunk 2]
```diff
@@ -940,7 +940,7 @@ inputs:
     value: 'b'
 ```
 
-Then you can use `abc templates golden-test` to record or verify the tests.
+Then you can use `abc templates golden-test` to record(capture the anticipated outcome akin to expected output in unit test)or verify the tests.
```
Comment (drevell) [inline README.md:943]: Same here with space before parenthesis

File: templates/commands/goldentest/record.go [hunk 3]
```diff
@@ -37,17 +37,17 @@ type RecordCommand struct {
 }
 
 func (c *RecordCommand) Desc() string {
-	return "record the template rendering result to golden tests"
+	return "record the template rendering result to golden tests(capture the anticipated outcome akin to expected output in unit test)"
```
Comment (drevell) [inline templates/commands/goldentest/record.go:40]: Same here and below with the space before the parenthesis

**Review by drevell — CHANGES_REQUESTED** [review]
Thanks for clarifying these docs

File: README.md (hunk 1 above)
Comment (capri-xiyue) [reply README.md:922]: done.

Comment (capri-xiyue) [reply README.md:922]: done

Comment (capri-xiyue) [reply README.md:922]: done

File: README.md (hunk 2 above)
Comment (capri-xiyue) [reply README.md:943]: done

File: templates/commands/goldentest/record.go (hunk 3 above)
Comment (capri-xiyue) [reply templates/commands/goldentest/record.go:40]: done

File: README.md [hunk 4]
```diff
@@ -940,7 +941,7 @@ inputs:
     value: 'b'
 ```
 
-Then you can use `abc templates golden-test` to record or verify the tests.
+Then you can use `abc templates golden-test` to record (capture the anticipated outcome akin to expected output in unit test)or verify the tests.
```
Comment (drevell) [inline README.md:944]: I still see this line as being 145 characters long

File: templates/commands/goldentest/record.go [hunk 5]
```diff
@@ -37,17 +37,17 @@ type RecordCommand struct {
 }
 
 func (c *RecordCommand) Desc() string {
-	return "record the template rendering result to golden tests"
+	return "record the template rendering result to golden tests (capture the anticipated outcome akin to expected output in unit test)"
```
Comment (drevell) [inline templates/commands/goldentest/record.go:40]: These lines are also quite long

**Review by drevell — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (capri-xiyue) [reply README.md:944]: done

File: templates/commands/goldentest/record.go (hunk 5 above)
Comment (capri-xiyue) [reply templates/commands/goldentest/record.go]: done

**Review by drevell — APPROVED** [review]

**Pull request merged**